filebeat/module/elasticsearch/audit/ingest/pipeline-json.yml

```diff
@@ -3,6 +3,11 @@ processors:
   - json:
       field: message
       target_field: elasticsearch.audit
+  - dot_expander:
+      field: event.type
+      path: elasticsearch.audit
+  - drop:
+      if: '!["rest", "transport", "ip_filter", "security_config_change"].contains(ctx.elasticsearch?.audit?.event?.type)'
```

Comment:
it looks correct, but what if ES starts emitting additional event.type? At the same time I don't know how else we can distinguish.. Maybe if we make the condition more complex. <7.3 the if we have now. >7.3 we would check against event.type = audit?

Reply:
I agree that current condition is not the perfect one, but couldn't come up with something better.
>  Maybe if we make the condition more complex. <7.3 the if we have now. >7.3 we would check against event.type = audit?

is there any unified way to define elasticsearch version?
in my setup I see only `kubernetes.container.name` and `container.image.name` that contains version of image - but it is too specific to the environment where elasticsearch is running

Reply:
@pgomulka I've rethought this condition a bit, do you think it could be a better option:
```
  - drop:
      if: 'ctx.elasticsearch.audit.containsKey('type') && ctx.elasticsearch.audit.type != 'audit'
  - drop:
      if: '!ctx.elasticsearch.audit.containsKey('type') && !["rest", "transport", "ip_filter", "security_config_change"].contains(ctx.elasticsearch?.audit?.event?.type)'
```
?

Reply:
it looks good! should fit both scenarios well and be flexible for new event types
thanks!